contracts/System.sol

```diff
@@ -80,4 +94,104 @@ contract System {
   /// @param lowerBound requested lower bound of the param.
   /// @param upperBound requested upper bound of the param
   error OutOfBounds(string name, uint256 given, uint256 lowerBound, uint256 upperBound);
+
+  function init() external onlyNotInit {} // placeholder
+
+  function _gasprice() internal virtual view returns (uint) {
+    return tx.gasprice;
+  }
+
+  function _coinbase() internal virtual view returns (address) {
+    return block.coinbase;
+  } 
+
+  function govHub() external returns (address) {
+    return safe_govHubAddr();
+  }
+  // -------
+
+  function secureSend_(address sendTo, uint256 amount) internal returns (bool) {
+    if (!sufficientBalance_(amount)) {
+      return false;
+    }
+    return payable(sendTo).send(amount);
+  }
+
+  function secureTransfer_(address sendTo, uint256 amount) internal {
+    require(sufficientBalance_(amount), "insufficient balance");
+    payable(sendTo).transfer(amount);
+  }
+
+  function sufficientBalance_(uint256 amount) private view returns (bool) {
+    return address(this).balance >= amount;
+  }
+
+
+
+
+  // --- Registry functions below are platform-contracts that are safe to access ---
+
+  function safe_foundationPayable() internal returns(address payable) {
```

Comment:
All platform contracts are 'safe_' in the sense that we can rely on their trusted behavior with no additional checks/protection (Reentrry, DoS, the lot..).  Other addresses should at least be treated - I would suggest also prefixed - as 'unsafe'